tools/toolchain/clang_lto_pgo.sh

```diff
@@ -0,0 +1,108 @@
+#!/bin/bash -uex
+
+if [ -z "${CLANG_BUILD}" ]; then
+	echo "Skip building optimized clang"
+	exit 0
+elif [ "${CLANG_BUILD}" = "SKIP" ]; then
+        echo "Skip building optimized clang"
+        exit 0
+elif [ "${CLANG_BUILD}" = "BUILD" ]; then
+	echo "build optimized clang"
+elif [ "${CLANG_BUILD}" = "INSTALL" ]; then
+	echo "build and install optimized clang"
```

Comment:
I don't see why we'd want to build but not install.


On the other hand, I asked for a COPY option to copy the installed compiler from a previous toolchain.

Reply:
BUILD is supposed to keep somewhere that eventually will be exposed to the host (I could use /optimized-clang), so others can use it outside dbuild. It's incomplete - as the build is not static.

Reply:
I remember seeing work going into clang to enable static builds recently. May need to trawl the commit log.

Reply:
https://github.com/ClangBuiltLinux/tc-build/issues/150

Trying with LLVM_STATIC_LINK_CXX_STDLIB, let's see what we get.

Reply:
```
[ykaul@ykaul scylladb]$ ldd ~/scratch/clang-15 
	linux-vdso.so.1 (0x00007ffcd88a5000)
	libm.so.6 => /lib64/libm.so.6 (0x00007f0bd186c000)
	libz.so.1 => /lib64/libz.so.1 (0x00007f0bd1852000)
	libzstd.so.1 => /lib64/libzstd.so.1 (0x00007f0bc954d000)
	libstdc++.so.6 => /lib64/libstdc++.so.6 (0x00007f0bc9200000)
	libgcc_s.so.1 => /lib64/libgcc_s.so.1 (0x00007f0bd1832000)
	libc.so.6 => /lib64/libc.so.6 (0x00007f0bc9023000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f0bd1964000)
```

Reply:
It removed the dependency on llvm itself, but can it be made fully static?

Alternatively, we can try to package it as a Nix package.